subprojects/packagefiles/cfitsio/meson.build

```diff
@@ -0,0 +1,270 @@
+project(
+  'cfitsio',
+  'c',
+  version: '4.5.0',
+  meson_version: '>=1.1.0',
+  license: 'CFITSIO',
+  license_files: ['licenses/License.txt'],
+)
+has_fortran = add_languages(
+  'fortran',
+  native: false,
+  required: get_option('fortran'),
+)
+
+SOVERSION = 10
+
+cc = meson.get_compiler('c')
+libm = cc.find_library('m')
+
+if target_machine.system() != 'windows'
+  curl = dependency(
+    'libcurl',
+    required: get_option('curl'),
+    fallback: ['curl', 'libcurl'],
+  )
+  if curl.found()
+    add_project_arguments('-DCFITSIO_HAVE_CURL', language: 'c')
+  endif
+endif
+
+threads = dependency('threads', required: get_option('reentrant'))
+if threads.found()
+  add_project_arguments('-D_REENTRANT', language: 'c')
+endif
+
+zlib = dependency('zlib', fallback: ['zlib', 'zlib_dep'])
+
+if target_machine.system() != 'windows'
+  bzip2 = dependency(
+    'bzip2',
+    required: get_option('bzip2'),
+    fallback: ['bzip2', 'bzip2'],
+  )
+  if bzip2.found()
+    add_project_arguments('-DHAVE_BZIP2=1', language: 'c')
+  endif
+endif
+
+if get_option('hera')
+  add_project_arguments('-DBUILD_HERA=1', language: 'c')
+endif
+
+if cc.get_id() != 'msvc' and cc.has_function('gethostbyname') and cc.has_function(
+  'connect',
+)
+  add_project_arguments('-DHAVE_NET_SERVICES', language: 'c')
+  if cc.has_header_symbol('stdio.h', 'fmemopen')
+    add_project_arguments('-DHAVE_FMEMOPEN', language: 'c')
+  endif
+endif
+
+if get_option('sse2')
+  if cc.has_argument('-msse2')
+    add_project_arguments('-msse2', language: 'c')
```

Comment:
This seems like an odd thing to have as a project option. Can't users just compile with -msse2 themselves, especially given its often the default for a given -march?

Reply:
I was copying from the autotools/cmake file, but you are right this should be removed